### PR TITLE
fix: Check for ifHCInOctets and ifHighSpeed before falling back to if…

### PR DESCRIPTION
**Pull request by laf**
…Entry

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #6372

### CODE REVIEW

**Comment by murrant** [comment]
#6257

**Comment by laf** [comment]
Anything specific from that issue? This won't fix the issue you see there though.

**Comment by murrant** [comment]
It seems like it will.  Sorry, the title is misleading on that issue.

Basically, the device doesn't have ifHighSpeed, but ifSpeed was never walked by LibreNMS.

**Comment by laf** [comment]
So his fixes if we don't have both ifHSpeed and ifHCInOctets so if that's the case then yes it will.

File: includes/polling/ports.inc.php [hunk 1]
```diff
@@ -199,7 +199,8 @@
             $port_stats = snmpwalk_cache_oid($device, 'ifXEntry', $port_stats, 'IF-MIB');
         }
         $hc_test = array_slice($port_stats, 0, 1);
-        if (!isset($hc_test[0]['ifHCInOctets']) && !is_numeric($hc_test[0]['ifHCInOctets'])) {
+        if ((!isset($hc_test[0]['ifHCInOctets']) && !is_numeric($hc_test[0]['ifHCInOctets'])) &&
```
Comment (murrant) [inline includes/polling/ports.inc.php]: Are you sure you don't want `||` here?  When you use or, it fixes devices missing ifHighSpeed...

Comment (laf) [inline includes/polling/ports.inc.php]: We can switch to `||`.

Comment (laf) [inline includes/polling/ports.inc.php]: Pushed

**Pull request merged**